### PR TITLE
refactor(x/mint): introduce distributeDeveloperRewards methods; unit tests (1/2)

### PR DESCRIPTION
**Pull request by p0mvn**
Part of: https://github.com/osmosis-labs/osmosis/issues/1916

## What is the purpose of the change

Splitting up https://github.com/osmosis-labs/osmosis/pull/2026  [per suggestion](https://github.com/osmosis-labs/osmosis/pull/2026#issuecomment-1184427409).

The dev rewards distribution logic is 1:1 copied from how it used to be and encapsulated into a function. Additionally, unit tested.

## Testing and Verifying

This change added tests and can be verified as follows:

TestDistributeDeveloperRewards

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: x/mint/keeper/keeper.go [hunk 1]
```diff
@@ -198,98 +198,126 @@ func (k Keeper) DistributeMintedCoin(ctx sdk.Context, mintedCoin sdk.Coin) error
...
+	}
+
+	// call an hook after the minting and distribution of new coins
+	k.hooks.AfterDistributeMintedCoin(ctx, mintedCoin)
+
+	return err
+}
+
+// distributeToModule distributes mintedCoin multiplied by proportion to the recepientModule account.
+func (k Keeper) distributeToModule(ctx sdk.Context, recipientModule string, mintedCoin sdk.Coin, proportion sdk.Dec) (sdk.Int, error) {
```
Comment (p0mvn) [inline x/mint/keeper/keeper.go:233]: This function is not new and has not been changed. The diff is off

**Review by ValarDragon — APPROVED** [review]
Looks great, thanks!

File: x/mint/keeper/keeper.go (hunk 1 above)
Comment (ValarDragon) [reply x/mint/keeper/keeper.go:233]: Yeah lol, this is the classic git diff resolution being awful https://twitter.com/valardragon/status/1542516631481462785?s=20&t=XAD3fi6DjOdJaSvvFAU4EQ

Theres got to be some sort of browser extension or vscode extension to make this easier :/

Comment (p0mvn) [reply x/mint/keeper/keeper.go:233]: I heard that some folks like using Beyond Compare locally by setting it up as a Git diff tool: https://www.scootersoftware.com/ 

Not sure how / if it is better though

**Pull request merged**